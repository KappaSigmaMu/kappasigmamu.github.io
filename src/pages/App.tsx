import { Suspense } from 'react'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { AccountContextProvider } from '../account/AccountContext'
import { Navbar } from '../components/Navbar'
import { KusamaContextProvider } from '../kusama'
import { GlobalStyle } from '../styles/globalStyle'
import { Theme } from '../styles/Theme'
import { CyborgGuide } from './CyborgGuide'
import { Bids } from './home/Bids'
import { Home } from './home/Home'
import { LandingPage } from './LandingPage'
import { Welcome } from './Welcome'

const Main = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={
          <>
            <Navbar showSocialIcons showAccount />
            <LandingPage />
          </>
        }/>
        <Route path="/guide" element={
          <>
            <Navbar showBrandIcon showSocialIcons showAccount />
            <CyborgGuide />
          </>
        }/>
        <Route path="/welcome" element={
          <>
            <Navbar showBrandIcon showGalleryButton showAccount />
            <Welcome />
          </>
        }/>
        <Route path="/home" element={
          <>
            <Navbar showBrandIcon showGalleryButton showAccount />
            <Home />
          </>
        }/>
        <Route path="/home/bids" element={
          <>
            <Navbar showBrandIcon showGalleryButton showAccount />
            <Bids />
          </>
        }/>
      </Routes>
    </BrowserRouter>
  )
}

const App = () => (
  <>
    <GlobalStyle />
    <KusamaContextProvider>
      <AccountContextProvider>
        <ThemeProvider theme={Theme}>
          <GlobalStyle />
            <Suspense fallback={<p>ERROR/LOADING...</p>}>
              <Main />
            </Suspense>
        </ThemeProvider>
      </AccountContextProvider>
    </KusamaContextProvider>
  </>
)

export { App }
