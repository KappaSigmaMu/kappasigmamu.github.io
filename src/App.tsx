import { Container, Nav, Navbar, Button } from 'react-bootstrap'
import { Switch, Route, Link } from 'react-router-dom'
import NavLogo from '../src/static/nav-logo.png'
import { About } from './pages/About'
import { Blog } from './pages/Blog'
import { CyborgJourney } from './pages/CyborgJourney'
import { Home } from './pages/Home'

function App() {
  return (
    <Container>
      <Navbar bg="black" variant="dark">
        <Container fluid>
          <Navbar.Brand as={Link} to="/">
            <img
              src={NavLogo}
              alt="KappaSigmaMu"
              width="150"
              className="d-inline-block align-top"
            />
          </Navbar.Brand>
          <Nav className="mx-auto">
            <Nav.Link as={Link} to="/">
              Kusama Network
            </Nav.Link>
            <Nav.Link as={Link} to="/about">
              About
            </Nav.Link>
            <Nav.Link as={Link} to="/blog">
              Blog
            </Nav.Link>
            <Nav.Link as={Link} to="/cyborg-journey">
              Cyborg Journey
            </Nav.Link>
          </Nav>
          <Button>Connect Wallet</Button>
        </Container>
      </Navbar>

      <Switch>
        <Route exact path="/">
          <Home />
        </Route>
        <Route path="/about">
          <About />
        </Route>
        <Route path="/blog">
          <Blog />
        </Route>
        <Route path="/cyborg-journey">
          <CyborgJourney />
        </Route>
      </Switch>
    </Container>
  )
}

export { App }
