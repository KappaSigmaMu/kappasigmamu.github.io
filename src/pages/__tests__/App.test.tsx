import { BrowserRouter } from 'react-router-dom'
import ShallowRenderer from 'react-test-renderer/shallow'
import { App } from '../App'

test('renders App', () => {
  const renderer = new ShallowRenderer()

  const tree = renderer.render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  )

  expect(tree).toMatchSnapshot()
})
