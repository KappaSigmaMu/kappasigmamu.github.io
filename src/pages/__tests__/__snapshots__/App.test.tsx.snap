// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "back": [Function],
      "block": [Function],
      "createHref": [Function],
      "forward": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <App />
</Router>
`;
